src/shared/organisms/DataExplorerGraphFlowNavigationStack/NavigationStack.spec.tsx

```diff
@@ -0,0 +1,260 @@
+    title: 'Allen Institute for Brain Science',
+    types: ['Agent', 'Organization'],
+    resource: [
+      'bbp',
+      'atlas',
+      'https://www.grid.ac/institutes/grid.417881.3',
+      1,
+    ],
+  },
+  links: [
+    {
+      _self:
+        'https://bbp.epfl.ch/nexus/v1/resources/bbp/lnmce/datashapes:dataset/traces%2F460bfa2e-cb7d-4420-a448-2030a6bf4ae4',
+      title: '001_141216_A1_CA1py_R_MPG',
+      types: ['Entity', 'Trace', 'Dataset'],
+      resource: [
+        'bbp',
+        'lnmce',
+        'https://bbp.epfl.ch/neurosciencegraph/data/traces/460bfa2e-cb7d-4420-a448-2030a6bf4ae4',
+      ],
+    },
+    {
+      isDownloadable: false,
+      _self:
+        'https://bbp.epfl.ch/nexus/v1/resources/bbp/atlas/datashapes:organization/https:%2F%2Fwww.grid.ac%2Finstitutes%2Fgrid.5333.6',
+      title: 'Ecole Polytechnique Federale de Lausanne',
+      types: ['Organization', 'prov#Agent'],
+      resource: [
+        'bbp',
+        'atlas',
+        'https://www.grid.ac/institutes/grid.5333.6',
+        1,
+      ],
+    },
+  ],
+  shrinked: false,
+  highlightIndex: -1,
+  limited: false,
+};
+const fourthItemInStack = {
+  isDownloadable: false,
+  _self:
+    'https://bbp.epfl.ch/nexus/v1/resources/neurosciencegraph/datamodels/datashapes:ontologyentity/https:%2F%2Fbbp.epfl.ch%2Fontologies%2Fcore%2Fefeatures%2FAHPAmplitude',
+  title: 'AHP Amplitude',
+  types: ['Class'],
+  resource: [
+    'neurosciencegraph',
+    'datamodels',
+    'https://bbp.epfl.ch/ontologies/core/efeatures/AHPAmplitude',
+    28,
+  ],
+};
+describe('NavigationStack', () => {
+  let app: JSX.Element;
+  let component: RenderResult;
+  let container: HTMLElement;
+  let rerender: (ui: React.ReactElement) => void;
+  let store: Store<any, AnyAction>;
+  let user: UserEvent;
+  beforeAll(() => {
+    const history = createMemoryHistory({});
+
+    const nexus = createNexusClient({
+      fetch,
+      uri: deltaPath(),
+    });
+    store = configureStore(
+      history,
+      { nexus },
+      { dataExplorer: initialDataExplorerState }
+    );
+    app = (
+      <Provider store={store}>
+        <Router history={history}>
+          <NexusProvider nexusClient={nexus}>
+            <NavigationStack />
+          </NexusProvider>
+        </Router>
+      </Provider>
+    );
+    component = render(app);
+    container = component.container;
+    rerender = component.rerender;
+    user = userEvent.setup();
+  });
+  it('should render the correct number of NavigationStackItem components in the state', () => {
+    const navigationItems = container.querySelectorAll(
+      '.navigation-stack-item:not(.no-more)'
+    );
+    expect(navigationItems.length).toBe(2);
+  });
+  it('should render the correct number of NavigationStackItem components after hit the return back btn', () => {
+    store.dispatch(ReturnBackDataExplorerGraphFlow());
```

Comment:
It would be better to get the button in the `NavigationBack` component here and click that instead of manually dispatching an action. This way we test user bahavior (i.e. clicking the button resets the stack) instead of implementation detail (dispatching this particular action resets the stack).